### ISSUE TITLE
Can't see Gcode Previews (Solved)

### ISSUE THREAD

**Issue by GitUser999A** [alexqzd/Marlin#10]
Hello,

Windows 11 Pro 21H2
Cura 4.12.1
Python 3.10.1
Flashed my Voxelab Aquila G32 with Default-NoProbe-G32.bin and Alex's Firmware v1.3.5.1 Beta 4.
Control -> Visual -> Show file Thumbnails is checked
![WhatsApp Image 2022-01-02 at 19 38 36](https://user-images.githubusercontent.com/22965744/147884395-258d9df1-798b-4b1c-b6f6-6d9e1b5f424b.jpeg)

When I slice something I can see that my .gcode files now includes base64 image:
![image](https://user-images.githubusercontent.com/22965744/147884228-b9b58446-0fa7-4b8b-969a-d92ce6e3a600.png)

Tried to restart printer. Enable/disable show file thumbnails. But still I can't get it to work.

**Comment by tititopher68-dev** [comment]
@GitUser999A 
Tu as pris le fichier gcode chez mriscoc ? C'est donc normal que cela ne fonctionne pas !

To include PrusaSlicer thumbnails:

Make a copy of the the printer profile, and go to menu>help>Show configuration folder.
Locate the printers subfolder and edit the .ini file of your printer.
Around the bottom (line 69), there is a line that says "`thumbnails =`" change it to "`thumbnails = 50x50,217x217`".
Then add PrusaSlicer_JPEG_Preview.py to the to the post-processing scripts section of print settings : 

1- Installation of Python (if not already installed) : `https://www.python.org/downloads/release/python-399/`
2- Copy the script '`PrusaSlicer_JPEG_Preview.py`' in a folder ie :  
`C:\Program Files\Prusa3D\Scripts_Post-processings\`
3- Open PrusaSlicer and go to Print Settings Tab, Post-processing scripts section and enter exactly in the input field `"C:\Program Files\Python39\python.exe" "C:\Program Files\Prusa3D\Scripts_Post-processings\PrusaSlicer_JPEG_Preview.py";`

![Config_Prusa_2](https://user-images.githubusercontent.com/84127160/146771361-385297e4-15f5-473a-8021-612939ae0165.jpg)

4- Next go to Printer Settings tab, Firmware section and check if you have `50x50, 217x217`. If not enter this parameter in the input field.

![Config_Prusa_1](https://user-images.githubusercontent.com/84127160/146771913-af5f0652-73f9-4062-a185-62a201de156f.jpg)

5- Don't forget to save your new profil.

To include Cura thumbnails : 

1- Go to the Help menu on Cura and select Show Configuration Folder in Windows you will get an explorer opened in the path inside of your user profile folder: `C:\Users\'user name'\AppData\Roaming\cura\`
2- Close Cura and save the script 'Cura_JPEG_Preview.py' in the scripts folder.
3- Open Cura, go to menu `Extensions -> Post Processing -> Modify G-Code` and press Add a script, then select the script  `Create JPEG Thumbnail` from the post processing plugin popup.
4- Select for Width and Height values : 50 and validate.
5- Again, go to menu `Extensions -> Post Processing -> Modify G-Code` and press Add a script, then select the script  `Create JPEG Thumbnail` from the post processing plugin popup.
6- And select for Width and Height values : 217 and validate.

Pictures from prusaslicer :  (from @alexqzd) : 
![foto1](https://user-images.githubusercontent.com/84127160/146775504-e323c193-20e4-4d67-88f3-c96abd6e8ba4.jpg)
![foto2](https://user-images.githubusercontent.com/84127160/146775519-7653370e-c54d-46c6-bdc8-e74588c7826e.jpg)



Nota : 
Malheureusement, si je ne sais pas si @alexqzd continue ou pas les fix concernant le Gcode Preview, car selon la taille du Preview, cela peut s'afficher ou pas

**Comment by GitUser999A** [comment]
Can someone give me file which includes thumbnail and works for you, so I can do some testing?

**Comment by tititopher68-dev** [comment]
@GitUser999A 
This is good but don't forget that if you have one or more Gcodes files with images weighing more than 5.5 kb, the preview part will not be displayed. The image is encoded in base 64. the weight of the image does not depend mainly on its 217x217 size, but depends greatly on the dimensions of the object to be displayed.
On the Jyers site, I modified the code to decrease the size to 180x180 and increasing the compression by 5%, which does not harm the display. And of course in the firmware code I also made the necessary changes. So the same Gcode encoded in 217x217 is not displayed, now is displayed in 180x180 ... Even with a very large object, I manage not to exceed the size of the maximum buffer.
